src/cache.rs

```diff
@@ -160,22 +160,50 @@ impl Cache {
         let platform = self.get_platform_dir();
 
         // Search for the page in the platform specific directory
-        if let Some(pf) = platform {
-            let path = platforms_dir.join(&pf).join(&page_filename);
-            if path.exists() && path.is_file() {
-                return Some(path);
+        let path = if let Some(pf) = platform {
+            let p = platforms_dir.join(&pf).join(&page_filename);
+            if p.exists() && p.is_file() {
+                Some(p)
+            } else {
+                None
             }
-        }
+        } else {
+            None
+        };
 
         // If platform is not supported or if platform specific page does not exist,
         // look up the page in the "common" directory.
-        let path = platforms_dir.join("common").join(&page_filename);
+        let path = if path.is_none() {
```

Comment:
Don't accept this yet, just noticed a bug here, will fix tomorrow. Path is overriding to none when the previous had Some. This is incorrect.

Reply:
Fixed